### PR TITLE
fix #1050 and introduce money ns that uses bignumber for arithmetic

### LINKED ISSUE #1050: App does not respond to any tap if try to send requested sum with 15 digits after comma
### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*:  Other account is requesting 0.111122223333444. Account from iOS device taps on request and taps on /send command. After this nothing happens. Pending transaction screen is not shown. Tap on any element has no effect, only scrolling works. 

#### Expected behavior
[comment]: # (Describe what you expected to happen.)
When user taps on /send then Pending transactions screen is shown
#### Actual behavior
[comment]: # (Describe what actually happened.)
Pending transactions screen is not shown. Tap on any button has no effect
### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)
video: https://drive.google.com/open?id=0Bz3t9zSg1wb7Mm82T1VraU1keTA
- From another device: request 0.111122223333444 from iOS device
- As iOS device: tap on request 
- Tap on send button

### Additional Information
[comment]: # (Please do your best to fill this out.)
* Status version:  0.9.6d64, 0.9.7 and 0.9.8
* Operating System: only iOS. Can reproduce on iPhone 7 plus with iOS 10.3.2. Can't reproduce on iPhone 6s with iOS 10.2.1

Console Log from iPhone 7 plus: 
[consolelog2.txt](https://github.com/status-im/status-react/files/1049542/consolelog2.txt)

### PR DESCRIPTION
**Pull request by oskarth**
fixes #1050 

Introduce a money namespace that solves a bunch of issues related to significant numbers being preserved and converted correctly.

### Summary:
The BigNumber version included in web3 sometimes hangs when dividing large numbers. Hence we want to use these functions instead of fromWei etc, which come bundled with web3. See
https://github.com/MikeMcl/bignumber.js/issues/120 for this regression being introduced in some JS environments. It is fixed in the MikeMcl/bignumber.js repo, but not in the web3 BigNumber fork:
https://github.com/ethereum/web3.js/issues/877

Additionally, while it is possible to use the BigNumber constructor without stringifying the number, this only works up to some 15 significant digits:
https://github.com/MikeMcl/bignumber.js/issues/120

Lastly, notice the bad rounding for native Javascript numbers above 17 digits that may result in errors earlier up the call chain. Ideally all money-related sensitive functions should be moved into this namespace to check for such matters:
(str 111122223333441239) => "111122223333441230"

### Steps to test:
- Open Status (iOS)
- Ensure you Ether on your account
- /send PERSON 0.111122223333444
- Notice that app no longer crashes

[comment]: # (PRs will only be accepted if squashed into single commit.)
status: ready

Tested on iOS simulator. As this change involves change of packages etc, and Web3 already includes their version of BigNumber, we have to be careful to test it on real devices to make sure the wrong version of the library isn't being called.

### CODE REVIEW

**Comment by janherich** [comment]
@oskarth @rasom Those changes are great and I'm actually quite puzzled that library like web3 (basically official JS gateway for ethereum blockchain) comes with such trivial bugs which are not already resolved.
Do you have some idea how to include correct BigNumber version for JailVM js environment where our bot js implementations are running ?

File: src/status_im/utils/money.cljs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+;; matters:
+;; (str 111122223333441239) => "111122223333441230"
+
+(defn bignumber [n]
+  (dependencies/BigNumber. (str n)))
+
+(def ether-unit-value (bignumber "1000000000000000000"))
+
+(defn wei->ether [n]
+  (.dividedBy (bignumber (str n)) ether-unit-value))
```
Comment (jeluard) [inline src/status_im/utils/money.cljs]: Any reason for the redundant `str` call?

File: test/cljs/status_im/test/utils/money.cljs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+(ns status-im.test.utils.money
+  (:require [cljs.test :refer-macros [deftest is]]
+            [status-im.utils.money :as money]))
+
+(deftest bignumber
+  (is (= (str (money/wei->ether 111122223333444000))
+         "0.111122223333444")))
+
+(deftest bignumber-2
```
Comment (jeluard) [inline test/cljs/status_im/test/utils/money.cljs]: A better error message would be generated by having a more descriptive test name (say `wei->ether`) then relying on `testing` macro to distinguish scenarios. A single test would be sufficient then.

File: src/status_im/utils/money.cljs (hunk 1 above)
Comment (oskarth) [inline src/status_im/utils/money.cljs]: Initially I didn't include it in the bignumber constructor, fixed now, thanks!

File: test/cljs/status_im/test/utils/money.cljs (hunk 2 above)
Comment (oskarth) [inline test/cljs/status_im/test/utils/money.cljs]: @jeluard  Good point, fixed!

**Pull request merged**